docs/docs/kds-configuration.md

```diff
@@ -207,32 +154,95 @@ kds {
+
+## KDS keys for mail
+The KDS is also capable of providing keys for use with [Conclave mail](mail.md) through use of a specially constructed [`PostOffice`](api/-conclave/com.r3.conclave.mail/-post-office/index.html).
+
+When using a KDS post office, the post office requests a public key from the KDS for use in a Conclave [`PostOffice`](api/-conclave/com.r3.conclave.mail/-post-office/index.html) and specifies which enclaves should be permitted access to the corresponding private key.
+Upon receiving mail, the enclave requests the private key from the same KDS and uses it to decrypt the mail object from the client.
+In this manner, users can create mail objects which can be decrypted by any enclave that matches a given set of constraints, without first having to receive and attest to the [`EnclaveInstanceInfo`](api/-conclave/com.r3.conclave.common/-enclave-instance-info/index.html) of a specific enclave.
+
+### Configuring the Enclave
+To make use of KDS keys for mail, the `kdsEnclaveConstraints` field in the enclaves `build.gradle` must be configured.
+This configuration alone is sufficient to use KDS keys with mail.
+
+```groovy
+...
+conclave {
+    kds {
+        kdsEnclaveConstraints = "<KDS enclave policy constraints>"
+    } 
+}
+...
+```
+
+!!!note
+For the R3 KDS, the following constraint should be used: `S:B4CDF6F4FA5B484FCA82292CE340FF305AA294F19382178BEA759E30E7DCFE2D PROD:1 SEC:STALE`.
+
+
+### Defining a KDS key spec
+To make use of KDS keys with mail, the client must construct an appropriately configured [`PostOffice`](api/-conclave/com.r3.conclave.mail/-post-office/index.html) object with which to encrypt mail.
+The first step is to create a [`KDSKeySpec`](api/-conclave/com.r3.conclave.common.kds/-k-d-s-key-spec/-k-d-s-key-spec.html) object which specifies how the public key should be derived and which enclaves should be allowed access to the corresponding private key:
+
+```java
+import com.r3.conclave.common.kds.KDSKeySpec;
+
+...
+var kdsKeySpec = new KDSKeySpec(
+        keyName,
+        masterKeyType,
+        policyConstraint);
+...
+```
+
+- `keyName` - The name of the key. This name is used during key derivation and if changed, will result in a different key being generated. This can be used to separate keys by purpose.
+- `masterKeyType` - The type of master key to use for the derivation (see [`MasterKeyType`](api/-conclave/com.r3.conclave.common.kds/-master-key-type/index.html)).
+- `policyConstraint` - A string which specifies the constraints that the enclave must meet in order to derive the corresponding private key and decrypt messages. The string should be in the [enclave constraints DSL](constraints.md) format.
+
+### Using the KDS post office builder
+Once an appropriate key spec has been created, a post office can be instantiated using the [`PostOfficeBuilder`](api/-conclave/com.r3.conclave.client/-post-office-builder/index.html) class.
```

Comment:
After creating an appropriate key spec, you can instantiate a post office using the [`PostOfficeBuilder`](api/-conclave/com.r3.conclave.client/-post-office-builder/index.html) class.